### PR TITLE
Show create button at fixed position at the bottom of the dropdown list

### LINKED ISSUE #35969: [Enhancement] Fix the create button at the bottom of the dropdown list
**Related to #35851 -- some work on #35851 deals with the AC in this issue**

We noticed in user testing that some people couldn't figure out how to add a new item in the dropdown component. In addition to the existing logic that allows users to type the name and select an option from the list to create a new item, we want to add a fixed button that will be visible at all times.

https://user-images.githubusercontent.com/79307566/207562708-5b87a0d5-40fd-4cdc-93fc-466cf5c875d6.mov

**Acceptance criteria**

- [ ] There's a new item fixed at the bottom of the list. It is visible to users as they scroll up and down the list.
- [ ] Users can quickly navigate to it using the `Tab` key.
- [ ] The button is `Gutenberg-Blue` (it's now `Gutenberg-900`) and says `+ Create new`.
- [ ] Clicking the button or pressing `Enter` opens the item creation modal.
- [ ] When users start typing, the button's label changes to `+ Create "[phrase]"` just like it does now.

### PR DESCRIPTION
**Pull request by nathanss**
### All Submissions:

-   [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Shows the Create button from the categories field at a fixed position at the bottom, so that the users can always see them.

Check additional acceptance criteria at issue #35969.

There is one acceptance criterion that I was unable to achieve and I think we should consider removing it in case we want to maintain the same behavior at the component, which is having the focus on the input field at all times:
> Users can quickly navigate to it using the Tab key.
It would involve at least preventing the default tab action. I'm not sure about doing this. The users can achieve similar behavior by pressing the End key on the keyboard.

Closes #35969 .

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Open the new product manager
2. Focus on the Category field
3. Check that "Create new" appears at the bottom of the list
4. Type some non-existing value
5. Check that the label changes to "Create 'value'"
6. Check that the up, down, home, and end buttons work as expected
7. Check that the "Create new" button is clickable

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [ ] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/product-editor/src/components/details-categories-field/category-field-add-new-item.tsx [hunk 1]
```diff
@@ -20,14 +20,11 @@ type CategoryFieldAddNewItemProps = {
 
 export const CategoryFieldAddNewItem: React.FC<
 	CategoryFieldAddNewItemProps
-> = ( { item, highlightedIndex, getItemProps, items } ) => {
+> = ( { item, highlightedIndex, items, getItemProps } ) => {
```
Comment (octaedro) [inline packages/js/product-editor/src/components/details-categories-field/category-field-add-new-item.tsx:23]: Out of curiosity, is there a reason for changing the order of these?

**Review by octaedro — APPROVED** [review]
Nice job @nathanss! This is testing well on my end and the code LGTM. Just left a small question but I'll pre-approve this PR.

**Comment by nathanss** [comment]
Hello @octaedro 

Thanks for the review, but I'm wondering if we should merge this since I'm working on [Migrate category field to TreeControl component](https://github.com/woocommerce/woocommerce/issues/37043) which would make this obsolete.  Sorry I didn't set this as blocked.

**Comment by octaedro** [comment]
Got it! I'm going to close this PR. Thank you @nathanss 🙂

**Pull request closed without merging**